coffea/nanoevents/mapping/parquet.py

```diff
@@ -6,18 +6,29 @@
 import json
 from coffea.nanoevents.mapping.base import UUIDOpener, BaseSourceMapping
 from coffea.nanoevents.util import quote, key_to_tuple, tuple_to_key
+import pyarrow.dataset as ds
 
 
 # IMPORTANT -> For now the uuid is just the uuid of the pfn.
 #              Later we should use the ParquetFile common_metadata to populate.
 class TrivialParquetOpener(UUIDOpener):
     class UprootLikeShim:
-        def __init__(self, file):
+        def __init__(self, file, shim_options):
             self.file = file
+            self.shim_options = shim_options
+            self.schema = shim_options['schema_arrow']
+            self.format = shim_options['format']
 
         def read(self, column_name):
             # make sure uproot is single-core since our calling context might not be
-            return self.file.read([column_name], use_threads=False)
+            if self.format == 'rados-parquet':
+                return ds.dataset(
```

Comment:
If we're re-constructing the dataset each column access, doesn't this mean a whole connection setup and teardown? We should construct the parquet dataset object once in the constructor, or even back in the NanoEventsFactory

Reply:
Yeah, very sorry for that. I realize I was kind of ignorant of the performance and code quality and focussed more on getting the stuff to work. Will look for more such issues.